sssd_test_framework/utils/sssd.py

```diff
@@ -247,6 +262,11 @@ def async_restart(
         if service == "sssd":
             self.svc.async_stop("sssd-kcm.service")
 
+        if clean and service == "sssd":
+            self.svc.stop(service)
+            self.clear()
+            return self.svc.async_restart(service)
```

Comment:
Just clear is sufficient as in other places, there is no need to stop it explicitly. It will also break the async purpose of this method.